internal/v4/api.go

```diff
@@ -266,36 +263,39 @@ func (h *Handler) entityExists(id *charm.Reference, req *http.Request) (bool, er
 }
 
 func (h *Handler) baseEntityQuery(id *charm.Reference, selector map[string]int, req *http.Request) (interface{}, error) {
-	var val mongodoc.BaseEntity
-	id1 := *id
-	id1.Series = ""
-	id1.Revision = -1
-	err := h.store.DB.BaseEntities().
-		Find(bson.D{{"_id", &id1}}).
-		Select(selector).
-		One(&val)
-	if err == mgo.ErrNotFound {
+	fields := make([]string, 0, len(selector))
```

Comment:
Yes, we could possibly clean that up, but it will require a change of interface somewhere. I'm going to leave this for another PR.